### PR TITLE
[Compiler-v2] handle the case Seq(exp) iteratively in translate_seq_recursively

### PR DESCRIPTION
**Pull request by rahxephon89**
## Description

This PR fixes https://github.com/aptos-labs/aptos-core/issues/13250 by using iteration to handle Seq(exp) in the function `translate_seq_recursively`.

Note that this PR basically tries to resolve the same issue that the PR #13436 but uses a different approach. Not sure which ways to go.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Existing tests pass and the test third_party/move/move-bytecode-verifier/transactional-tests/tests/reference_safety/call_function_with_many_acquires does not overflow with MOVE_COMPILER_V2 set to true.

Also tested by the following script:

```
for value in false true; do 
UB=1 MOVE_COMPILER_V2=$value cargo test \
      -p "aptos-transactional-test-harness" \
      -p "bytecode-verifier-transactional-tests" \
      -p "move-async-vm" \
      -p "move-cli" \
      -p "move-model" \
      -p "move-package" \
      -p "move-prover-bytecode-pipeline" \
      -p "move-stackless-bytecode" \
      -p "move-to-yul" \
      -p "move-transactional-test-runner" \
      -p "move-unit-test"
      -p "move-vm-transactional-tests" \
      --no-fail-fast \
      >& test-$value.out
done
```

## Key Areas to Review

Whether the logic in iteration is correct.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-bytecode-verifier/transactional-tests/tests/type_safety/type_error_after_branch.v2_exp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+processed 1 task
```
Comment (rahxephon89) [inline third_party/move/move-bytecode-verifier/transactional-tests/tests/type_safety/type_error_after_branch.v2_exp:1]: This file is consistent with third_party/move/move-bytecode-verifier/transactional-tests/tests/type_safety/type_error_after_branch.exp

**Review by brmataptos — APPROVED** [review]
LGTM, thanks.

**Review by wrwg — APPROVED** [review]

File: third_party/move/move-model/src/builder/exp_builder.rs [hunk 2]
```diff
@@ -2762,38 +2762,45 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                     self.exit_scope();
                     self.new_bind_exp(loc, pat, binding, rest.into_exp())
                 },
-                Seq(exp) if items.len() > 1 => {
+                Seq(_) if items.len() > 1 => {
                     // This is an actual impl language sequence `s;rest`.
```
Comment (wrwg) [inline third_party/move/move-model/src/builder/exp_builder.rs:2766]: PS. Perhaps pull this entire block into its own function. (Easy with RustRover.)

Comment (rahxephon89) [reply third_party/move/move-model/src/builder/exp_builder.rs]: done

**Pull request merged**